### PR TITLE
Shader blending in Vulkan through input attachments

### PR DESCRIPTION
**Pull request by hrydgard**
This uses input attachments, a subpass self-dependency and barriers to implement limited shader blending for unextended Vulkan, avoiding copies.

This of course has the limitation that self-overlapping draw calls will not necessarily (although might on some hardware or if lucky) blend correctly with earlier parts of themselves, but it's generally not an issue and is also an issue of the copy-texture-based shader blending (though the wrong results will likely not be exactly the same).

Should be a modest performance boost for some games, and result in cleaner output in RenderDoc. And is simply the right thing to do, additionally opens up to more uses for this technique, such as implementing the self-depal horror used in Jackass (#13676).

Things left to do:

- [x] Don't force this on in all render passes, since it forces us to use a possibly less optimal image layout (GENERAL instead of COLOR_ATTACHMENT_OPTIMAL). Fixing this will require a bit of refactoring and having separately compiled pipelines for these (since pipelines include renderpasses in their creation).
- [x] Testing on various hardware
  - [x] Works great on NVIDIA
  - [x] Works great on both Mali and Adreno in Outrun
  - [ ] Breaks on Adreno in DBZ: TTT :( (but works on Mali)

### CODE REVIEW

**Comment by hrydgard** [comment]
Rebased on master, confirmed working in Outrun etc for the logic stuff. Still, work remains before this can be merged.

File: Common/GPU/ShaderWriter.cpp [hunk 1]
```diff
@@ -10,7 +10,6 @@ const char * const vulkan_glsl_preamble_fs =
 "#version 450\n"
 "#extension GL_ARB_separate_shader_objects : enable\n"
 "#extension GL_ARB_shading_language_420pack : enable\n"
-"#extension GL_ARB_conservative_depth : enable\n"
```
Comment (unknownbrackets) [inline Common/GPU/ShaderWriter.cpp:13]: I thought we needed this for `writer.C("layout (depth_unchanged) out float gl_FragDepth;\n");`?  This is part of that Adreno/Mali bug with discards and stencil.

-[Unknown]

Comment (hrydgard) [reply Common/GPU/ShaderWriter.cpp:13]: That's right, just kept getting compiler warnings about it being unused, but of course should simply check whether it's needed before emitting it, or just ignore it. I'll fix it up when it's time to get this in.

**Comment by hrydgard** [comment]
Rebased on top of https://github.com/hrydgard/ppsspp/pull/15982 , which should be merged first.

**Comment by hrydgard** [comment]
Alright, rebased on master after merging #15982  . Will need more testing before merge.

**Review by unknownbrackets — APPROVED** [review]

**Comment by hrydgard** [comment]
This now works fine, except for some reason in Dragon Ball Z: TTT. I'm guessing there's some small overlap in the draw, where behavior becomes undefined (and accidentally worse than without this...). Bah.

**Comment by hrydgard** [comment]
Actually doesn't fix it, still getting artifacts along the diagonal. Seems this type of shader blending isn't working on at least old Adreno drivers. I'm gonna add a bug flag for now, and only enable this on non-Adreno.

**Comment by hrydgard** [comment]
OK, let's see if anything else breaks, hopefully not :)

**Pull request merged**